[snapshot] Sort screenshots.html by screen name

claesjacobsson opened the pull request:
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context

The screenshots.html page currently sorts the screenshots by language and device type. 
Sometimes it's useful to compare all localizations of a screen side by side. This PR makes that possible by providing two sorting options: _By Language_ and _By Screen_.

Fixes issue #13626.


### Description

_reports_generator.rb_ now produces two data objects (_data_by_language_ and _data_by_screen_) for _page.html.erb_ to use when creating _screenshots.html_.

_screenshots.html_ has been updated with a top menu bar to jump between the two sorting options.

As far as I understand, the current version only tests the device names being used in _reports_generator_. I have not made any changes and didn't see a reason to add any tests at this time.

claesjacobsson committed:
Merge remote-tracking branch 'fastlane/master'

claesjacobsson committed:
Merge remote-tracking branch 'fastlane/master'

claesjacobsson committed:
Merge remote-tracking branch 'fastlane/master'

claesjacobsson committed:
[snapshot] Added sorting by screen name in screenshots.html

Sorting the screenshots by screen name is useful when you want to
compare all localizations of a specific screen.

claesjacobsson committed:
[snapshot] Bold titles in screenshots.html

claesjacobsson committed:
Fixed image overlay info and scroll order

claesjacobsson committed:
Merge remote-tracking branch 'fastlane/master' into snapshot/sort-by-screen

janpio reviewed (approved):
Works as described, looks like this:

![image](https://user-images.githubusercontent.com/183673/50158715-2bf0c600-02d5-11e9-8858-3e6f31aa74f7.png)

The only thing that one could nitpick here is that it only works with Javascript enabled. In my testing, with Javascript disabled in Firefox, I couldn't see any screenshots at all. 

Maybe the tabs could both be visible by default, and only set to `display:none` via Javascript? (Add another headline above the "tab content" so it is clear what is actually shown below, and then also hide that new headline if you want to) What do you think?

claesjacobsson commented:
Didn't consider that. Your idea makes sense. Would you like me to make a separate PR for that or add to this one?

janpio commented:
If you can, just commit it to this PR's branch. Then we can merge it in one swoop.

claesjacobsson committed:
[snapshot] Give menu id 'sortMenu'

claesjacobsson committed:
[snapshot] Fix for browser with JS disabled

When Javascript is disabled:
- Show both ‘By Language’ and ‘By Screen’ content
- Hide sort menu and show extra title instead

claesjacobsson commented:
Done!

janpio commented on snapshot/lib/snapshot/page.html.erb:
```diff
@@ -64,23 +90,48 @@
+      <button class="tabLink" onclick="openTab(event, 'byScreen')">By Screen</button>
+    </div>
+    <div id="byLanguage" class="tabContent"><h1 class="tabTitle">By Language:</h1><% image_counter = 0 %><% @data_by_language.each do |language, content| %>
+      <h1 id="<%= language %>"><%= language %></h1>
```
I know this doesn't really matter here, but could we bump those down to h2 so the semantics are more correct? Thanks.

claesjacobsson replied:
For the language, you mean?

janpio replied:
yep. but actually, I'm fine either way.

claesjacobsson replied:
I can fix that. Makes sense.

claesjacobsson committed:
[snapshot] Use h2 tag for language and screen name

janpio reviewed (approved):
Looks perfect now, even without JS:

![image](https://user-images.githubusercontent.com/183673/50346385-86d32900-0532-11e9-9da6-d08342115d77.png)

And with JS:

![image](https://user-images.githubusercontent.com/183673/50346507-dc0f3a80-0532-11e9-886b-6e670dd5a0d7.png)
and
![image](https://user-images.githubusercontent.com/183673/50346498-d1ed3c00-0532-11e9-9f71-e4c732db1ab7.png)

janpio commented:
Thanks for applying my suggestions to this PR @claesjacobsson, the result is awesome! 🚀 💯
